openstack/messaging/v2/claims/requests.go

```diff
@@ -0,0 +1,76 @@
+package claims
+
+import (
+	"net/http"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// CreateOptsBuilder Builder.
+type CreateOptsBuilder interface {
+	ToClaimCreateRequest() (map[string]interface{}, string, error)
+}
+
+// CreateOpts params to be used with Create.
+type CreateOpts struct {
+	// The name of the queue to claim messages from.
+	QueueName string `json:"queue_name" required:"true"`
+
+	// Sets the TTL for the claim. When the claim expires un-deleted messages will be able to be claimed again.
+	TTL int `json:"ttl,omitempty"`
+
+	// Sets the Grace period for the claimed messages. The server extends the lifetime of claimed messages
+	// to be at least as long as the lifetime of the claim itself, plus the specified grace period.
+	Grace int `json:"grace,omitempty"`
+
+	// Set the limit of messages returned by create
+	Limit int `q:"limit,omitempty"`
+}
+
+// ToClaimCreateMap assembles a body and URL for a Create request based on
+// the contents of a CreateOpts.
+func (opts CreateOpts) ToClaimCreateRequest() (map[string]interface{}, string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	if err != nil {
+		return nil, q.String(), err
+	}
+
+	b, err := gophercloud.BuildRequestBody(opts, "")
+	if err != nil {
+		return b, "", err
+	}
+	if _, ok := b["Limit"]; ok {
```

Comment:
@jtopjian I don't know if this is the best way to accomplish what I am trying to do.

Reply:
Try the following struct field:

```
Limit int `q:"limit" json:"-"`
```

A few things about this:

1. `omitempty` is a core Go/JSON tag. `q` is an in-house Gophercloud tag. This catches everyone off-guard :)
2. Gophercloud's `q` support will automatically `omitempty` if the value is the zero-value: https://github.com/gophercloud/gophercloud/blob/master/params.go#L332
3. The special `json:"-"` tag _should_ drop a field from being included in the body: https://github.com/gophercloud/gophercloud/blob/master/params.go#L119-L121

If this doesn't work, then what you're doing here is the next best option :)